### ISSUE TITLE
feat: Wrapping math operations

### ISSUE THREAD

**Issue by Hyodar** [keep-starknet-strange/alexandria#214]
# Feature Request

**Describe the Feature Request**

It's actually pretty simple. Currently, wrapping operations require using the type-specific functions, such as `u64_wrapping_add`, which are not only type-specific but also result in code which is more difficult to understand. Plus, there's no direct operation for wrapping multiplications.

Wrapping operations can be pretty important in some implementations, and as such I feel like they need to have a friendlier interface soon. Even though I feel like this would be a feature corelib should have, I'm not sure what are the current priorities, so maybe it would be better to include it here first. Feel free to direct me to [cairo](https://github.com/starkware-libs/cairo) if you think we could push for adding it to corelib directly.

**Describe Preferred Solution**

A Rust-like interface for wrapping operations in integers.

Basically, adding `WrappingAdd`, `WrappingSub` and `WrappingMul` traits (or `WrappingOps`?) which provide, respectively, `wrapping_add`, `wrapping_sub` and `wrapping_mul`. Those will be implemented for all uints.

`Wrapping` as should be avoided so as to not cause any confusion related to Rust `Wrapping`.

**Describe Alternatives**

**Related Code**

Currently (considering a similar interface for `mul`):
```
u64_wrapping_mul(u64_wrapping_sub(u64_wrapping_add(value1, value2), value3), value4);
```

With this:
```
value1.wrapping_add(value2).wrapping_sub(value3).wrapping_mul(value4);
```

**Additional Context**

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes
- [ ] No

**Comment by enitrat** [comment]
Great idea! In Kakarot we're going to implement `CheckedOp` that returns Result<T,T> instead - but it's going to be done in a way similar to your idea.

**Comment by Hyodar** [comment]
Hey! Just added a quick PR for this.
